unity/Assets/Scripts/CanOpen_Object.cs

```diff
@@ -97,529 +78,218 @@ void Start ()
-                        if(openPositions[i].x - closedPositions[i].x != Mathf.Epsilon)
-                        {
-                            lerpToPosition.x = ((openPositions[i].x - closedPositions[i].x) * openPercentage) + closedPositions[i].x;
-                        }
-
-                        if(openPositions[i].y - closedPositions[i].y != Mathf.Epsilon)
-                        {
-                            lerpToPosition.y = ((openPositions[i].y - closedPositions[i].y) * openPercentage) + closedPositions[i].y;
-                        }
-
-                        if(openPositions[i].z - closedPositions[i].z != Mathf.Epsilon)
-                        {
-                            lerpToPosition.z = ((openPositions[i].z - closedPositions[i].z) * openPercentage) + closedPositions[i].z;
-                        }
-
-                        iTween.MoveTo(MovingParts[i], iTween.Hash(
-                        "position", lerpToPosition,
-                        "islocal", true,
-                        "time", animationTime,
-                        "easetype", "linear", "onComplete", "setisOpen", "onCompleteTarget", gameObject,
-                        "oncompleteparams", openPercentage));
-                    }
-
-                    else
-                    {
-                        //default to open position without percentage open modifiers
-                        lerpToPosition = openPositions[i];
-                        //some x, y, z components don't change when sliding open
-                        //only aply openPercentage modifier to components of vector3 that actually change
-                        if(openPositions[i].x - closedPositions[i].x != Mathf.Epsilon)
-                        {
-                            lerpToPosition.x = ((openPositions[i].x - closedPositions[i].x) * openPercentage) + closedPositions[i].x;
-                        }
-
-                        if(openPositions[i].y - closedPositions[i].y != Mathf.Epsilon)
-                        {
-                            lerpToPosition.y = ((openPositions[i].y - closedPositions[i].y) * openPercentage) + closedPositions[i].y;
-                        }
-
-                        if(openPositions[i].z - closedPositions[i].z != Mathf.Epsilon)
-                        {
-                            lerpToPosition.z = ((openPositions[i].z - closedPositions[i].z) * openPercentage) + closedPositions[i].z;
-                        }
-
-                        iTween.MoveTo(MovingParts[i], iTween.Hash(
-                        "position", lerpToPosition,
-                        "islocal", true,
-                        "time", animationTime,
-                        "easetype", "linear"));
-                    }
-
-				}
+        startOpenness = currentOpenPercentage;
+        for (int i = 0; i < MovingParts.Length; i++) {
+            Hashtable args = new Hashtable() {
```

Comment:
On the left, notice that all of these animations were nearly identical, except for passing in `rotation`, `position`, or `scale`. This `args` parameter just stores the identical arguments up-front, instead of explicitly writing them out for every single different open movement animation.